src/framework/radioButtons/RadioButtons.jsx

```diff
@@ -0,0 +1,101 @@
+
+import React, {
+  Component,
+} from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { Text } from '../framework';
+
+export default class RadioButtons extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedOption: this.props.selectedElement,
+    };
+
+    this.onSelectOption = this.onSelectOption.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.state.selectedOption !== this.props.selectedElement) {
+      // https://reactjs.org/docs/react-component.html#componentdidmount
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({
+        selectedOption: this.props.selectedElement
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.selectedOption !== this.props.selectedElement) {
+      // https://reactjs.org/docs/react-component.html#componentdidupdate
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        selectedOption: this.props.selectedElement
+      });
+    }
+  }
+
+  onSelectOption(element) {
+    this.setState({
+      selectedOption: element,
+    });
+    this.props.onSelect(element.value);
+  }
+
+  render() {
+    const onSelectCreator = value => () => this.onSelectOption(value);
```

Comment:
If we create the RadioElement component we wouldnt have to create this hacky wrapper